File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/Account.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

import java.util.Objects;

/**
 * Represents a bank account of a consumer. Contains holder information and different types of routing information.
 */
public class Account {

    private String holder;
    private Country country;
    private Sepa sepa;
    private Eft eft;
    private Ach ach;
    private Bacs bacs;

    /**
     * @return holder
     */
    public String getHolder() {
        return holder;
    }
    public void setHolder(String holder) {
        this.holder = holder;
    }
    public Account withHolder(String holder) {
        this.holder = holder;
        return this;
    }

    /**
     * @return country
     */
    public Country getCountry() {
        return country;
    }
    public void setCountry(Country country) {
        this.country = country;
    }
    public Account withCountry(Country country) {
        this.country = country;
        return this;
    }

    /**
     * @return sepa
     */
    public Sepa getSepa() {
        return sepa;
    }
    public void setSepa(Sepa sepa) {
        this.sepa = sepa;
    }
    public Account withSepa(Sepa sepa) {
        this.sepa = sepa;
        return this;
    }

    /**
     * @return eft
     */
    public Eft getEft() {
        return eft;
    }
    public void setEft(Eft eft) {
        this.eft = eft;
    }
    public Account withEft(Eft eft) {
        this.eft = eft;
        return this;
    }

    /**
     * @return ach
     */
    public Ach getAch() {
        return ach;
    }
    public void setAch(Ach ach) {
        this.ach = ach;
    }
    public Account withAch(Ach ach) {
        this.ach = ach;
        return this;
    }

    /**
     * @return bacs
     */
    public Bacs getBacs() {
        return bacs;
    }
    public void setBacs(Bacs bacs) {
        this.bacs = bacs;
    }
    public Account withBacs(Bacs bacs) {
        this.bacs = bacs;
        return this;
    }

    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Account account = (Account) o;
        return Objects.equals(this.holder, account.holder)
                && Objects.equals(this.country, account.country)
                && Objects.equals(this.sepa, account.sepa)
                && Objects.equals(this.eft, account.eft)
                && Objects.equals(this.ach, account.ach)
                && Objects.equals(this.bacs, account.bacs);
    }

    @Override
    public int hashCode() {
        return Objects.hash(holder, country, sepa, eft, ach, bacs);
    }

    @Override
    public String toString() {
        return "class Account {\n"
                + "        holder: " + toIndentedString(holder) + "\n"
                + "        country: " + toIndentedString(country) + "\n"
                + "        sepa: " + toIndentedString(sepa) + "\n"
                + "        eft: " + toIndentedString(eft) + "\n"
                + "        ach: " + toIndentedString(ach) + "\n"
                + "        bacs: " + toIndentedString(bacs) + "\n"
                + "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        return o == null
                ? "null"
                : o.toString().replace("\n", "\n        ");
    }
}

File: upl-purchase-sdk-js/src/unzer-paylater/model/Account.js
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

const ModelHelper = require("../util/ModelHelper");
const Ach = require("./Ach");
const Bacs = require("./Bacs");
const Country = require("./Country");
const Eft = require("./Eft");
const Sepa = require("./Sepa");

/**
 * Represents a bank account of a consumer. Contains holder information and different types of routing information.
 */
class Account {
    /**
     * @returns { String }
     */
    getHolder() {
        return this.holder;
    }
    /**
     * @param { String } holder
     */
    setHolder(holder) {
        this.holder = ModelHelper.validatePrimitive(holder, "string");
    }
    /**
     * @param { String } val
     */
    withHolder(val) {
        this.setHolder(val);
        return this;
    }

    /**
     * @returns { Country }
     */
    getCountry() {
        return this.country;
    }
    /**
     * @param { Country } country
     */
    setCountry(country) {
        this.country = ModelHelper.validateEnum(country, Country, "Country");
    }
    /**
     * @param { Country } val
     */
    withCountry(val) {
        this.setCountry(val);
        return this;
    }

    /**
     * @returns { Sepa }
     */
    getSepa() {
        return this.sepa;
    }
    /**
     * @param { Sepa } sepa
     */
    setSepa(sepa) {
        this.sepa = ModelHelper.validateObject(sepa, Sepa);
    }
    /**
     * @param { Sepa } val
     */
    withSepa(val) {
        this.setSepa(val);
        return this;
    }

    /**
     * @returns { Eft }
     */
    getEft() {
        return this.eft;
    }
    /**
     * @param { Eft } eft
     */
    setEft(eft) {
        this.eft = ModelHelper.validateObject(eft, Eft);
    }
    /**
     * @param { Eft } val
     */
    withEft(val) {
        this.setEft(val);
        return this;
    }

    /**
     * @returns { Ach }
     */
    getAch() {
        return this.ach;
    }
    /**
     * @param { Ach } ach
     */
    setAch(ach) {
        this.ach = ModelHelper.validateObject(ach, Ach);
    }
    /**
     * @param { Ach } val
     */
    withAch(val) {
        this.setAch(val);
        return this;
    }

    /**
     * @returns { Bacs }
     */
    getBacs() {
        return this.bacs;
    }
    /**
     * @param { Bacs } bacs
     */
    setBacs(bacs) {
        this.bacs = ModelHelper.validateObject(bacs, Bacs);
    }
    /**
     * @param { Bacs } val
     */
    withBacs(val) {
        this.setBacs(val);
        return this;
    }

    /**
     * @returns { Account }
     */
    static constructFromObject(data) {
        if (!data) return undefined;
        return new Account()
            .withHolder(ModelHelper.convertToType(data["holder"], String))
            .withCountry(ModelHelper.convertToType(data["country"], Country))
            .withSepa(ModelHelper.convertToType(data["sepa"], Sepa))
            .withEft(ModelHelper.convertToType(data["eft"], Eft))
            .withAch(ModelHelper.convertToType(data["ach"], Ach))
            .withBacs(ModelHelper.convertToType(data["bacs"], Bacs));
    }
}

module.exports = Account;
